py-polars/polars/dataframe/frame.py

```diff
@@ -4387,59 +4387,77 @@ def describe(
+        │ mean       ┆ 2.266667 ┆ 4.5      ┆ 0.666667 ┆ null ┆ null ┆ null       │
+        │ std        ┆ 1.101514 ┆ 0.707107 ┆ null     ┆ null ┆ null ┆ null       │
+        │ min        ┆ 1.0      ┆ 4.0      ┆ null     ┆ b    ┆ eur  ┆ 2020-01-01 │
+        │ 25%        ┆ 2.8      ┆ 4.0      ┆ null     ┆ null ┆ null ┆ null       │
+        │ 50%        ┆ 2.8      ┆ 5.0      ┆ null     ┆ null ┆ null ┆ 2021-07-05 │
+        │ 75%        ┆ 3.0      ┆ 5.0      ┆ null     ┆ null ┆ null ┆ null       │
+        │ max        ┆ 3.0      ┆ 5.0      ┆ null     ┆ c    ┆ usd  ┆ 2022-12-31 │
+        └────────────┴──────────┴──────────┴──────────┴──────┴──────┴────────────┘
         """
         if not self.columns:
             msg = "cannot describe a DataFrame without any columns"
             raise TypeError(msg)
 
         # Determine which columns should get std/mean/percentile statistics
-        stat_cols = {c for c, dt in self.schema.items() if dt.is_numeric()}
+        stat_cols, temporal_cols, bool_cols, numeric_result = set(), set(), set(), set()
+        schema = self.schema
+        for c, dt in schema.items():
+            if dt.is_numeric():
+                stat_cols.add(c)
+                numeric_result.add(c)
+            elif dt.is_temporal():
+                temporal_cols.add(c)
+            elif dt == Boolean:
+                bool_cols.add(c)
+                numeric_result.add(c)
+            elif dt == Null or dt.is_nested():
+                numeric_result.add(c)
 
         # Determine metrics and optional/additional percentiles
         metrics = ["count", "null_count", "mean", "std", "min"]
         percentile_exprs = []
         for p in parse_percentiles(percentiles):
             for c in self.columns:
-                expr = F.col(c).quantile(p) if c in stat_cols else F.lit(None)
+                if c in stat_cols:
+                    expr = F.col(c).quantile(p)
+                elif p == 0.5 and c in temporal_cols:
+                    expr = F.col(c).to_physical().median().cast(schema[c])
+                else:
+                    expr = F.lit(None)
                 expr = expr.alias(f"{p}:{c}")
                 percentile_exprs.append(expr)
             metrics.append(f"{p*100:g}%")
         metrics.append("max")
 
         mean_exprs = [
-            (F.col(c).mean() if c in stat_cols else F.lit(None)).alias(f"mean:{c}")
+            (
+                F.col(c).mean() if (c in stat_cols or c in bool_cols) else F.lit(None)
+            ).alias(f"mean:{c}")
             for c in self.columns
         ]
         std_exprs = [
             (F.col(c).std() if c in stat_cols else F.lit(None)).alias(f"std:{c}")
             for c in self.columns
         ]
-
         minmax_cols = {
             c
             for c, dt in self.schema.items()
             if not dt.is_nested()
-            and dt not in (Object, Null, Unknown, Categorical, Enum)
+            and dt not in (Object, Null, Unknown, Categorical, Enum, Boolean)
```

Comment:
> And it may be interesting if you run into all-False or all-True columns.

I can put it back (presumably it's cheap) but you get all of this information from `mean` now: you'd see `mean=0.0` or `mean=1.0` for those cases ;)

Reply:
That's true but it's still good to show it for consistency sake. If I see a null I might think that min/max are not implemented for the boolean type (that's what null means in the describe context).

Reply:
Done 👍